addons/purchase_stock/static/tests/tours/tour_helpers.js

```diff
@@ -0,0 +1,150 @@
+    return [
+        {
+            trigger: "body",
+            run: () => {
+                // Resolve luxon at runtime only, to prevent import error
+                luxon.DateTime.now = () => luxon.DateTime.fromFormat(jsDate, fmt);
+            },
+        },
+    ];
+}
+
+/**
+ * Sets the vendor on a purchase order (must already on PO).
+ * @param {string} vendorName An existing partner.
+ */
+export function selectPOVendor(vendorName) {
+    return [
+        {
+            content: "Fill Vendor Field on PO",
+            trigger: ".o_field_res_partner_many2one[name='partner_id'] input",
+            tooltipPosition: "bottom",
+            async run(actions) {
+                const input = this.anchor.querySelector("input");
+                await actions.edit(vendorName, input || this.anchor);
+            },
+        },
+        {
+            content: "Select vendor from many to onw",
+            isActive: ["auto"],
+            trigger: `.ui-menu-item > a:contains(${vendorName})`,
+            run: "click",
+        },
+    ];
+}
+
+/**
+ * Sets the WH on a purchase order (must already on PO).
+ * @param {string} warehouseName An existing warehouse.
+ */
+export function selectPOWarehouse(warehouseName) {
+    return [
+        {
+            content: "Fill Warehouse Field on PO",
+            trigger: ".o_field_many2one[name='picking_type_id'] input",
+            tooltipPosition: "bottom",
+            async run(actions) {
+                const input = this.anchor.querySelector("input");
+                await actions.edit(warehouseName, input || this.anchor);
+            },
+        },
+        {
+            content: "Select BaseWarehouse as PO WH",
+            isActive: ["auto"],
+            trigger: `.ui-menu-item > a:contains(${warehouseName})`,
+            run: "click",
+        },
+    ];
+}
+
+/**
+ * Sets the Suggest UI parameters
+ * @param {string} basedOn The label value of the "Replenish based on" select options (eg. "Last 3 months")
+ * @param {number} nbDays The value of the "Replenish for" input (eg. 90)
+ * @param {number} factor The value of the "x ...%" input (eg. 50)
+ * @param {number} wait Numebr of milliseconds before going to next step (eg. 1000).
+ * *I could find a workaround for the Timeout, because nothing on UI changes, just need to wait server round trip.
+ */
+export function setSuggestParameters({
+    basedOn = false,
+    nbDays = false,
+    factor = false,
+    wait = false,
+}) {
+    const steps = [];
+    if (basedOn) {
+        steps.push({
+            trigger: ".o_TimePeriodSelectionField",
+            async run(actions) {
+                await actions.click(this.anchor.querySelector(".o_select_menu_toggler"));
+                await new Promise((r) => setTimeout(r, 300));
+                await actions.click(`.o_popover .o_select_menu_item:contains('${basedOn}')`);
+            },
+        });
+    }
+    if (nbDays) {
```

Comment:
`nbDays` can be 0 which is falsy so the condition should handle that (same for `factor`)
```suggestion
    if (nbDays !== false) {
```